cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -75,51 +95,63 @@ inline daal::data_management::NumericTablePtr wrap_by_host_homogen_adapter(
     }
 }
 
-template <typename Float>
-inline daal::data_management::NumericTablePtr copy_to_daal_homogen_table(const table& table) {
-    auto rows = row_accessor<const Float>{ table }.pull();
-    return convert_to_daal_homogen_table(rows, table.get_row_count(), table.get_column_count());
-}
-
-template <typename Float>
+template <typename Data>
 inline daal::data_management::NumericTablePtr convert_to_daal_table(const homogen_table& table) {
-    auto wrapper = wrap_by_host_homogen_adapter(table);
-    if (!wrapper) {
-        return copy_to_daal_homogen_table<Float>(table);
-    }
-    else {
+    if (auto wrapper = wrap_by_host_homogen_adapter(table)) {
         return wrapper;
     }
+    return copy_to_daal_homogen_table<Data>(table);
 }
 
-template <typename Float>
+template <typename Data>
 inline daal::data_management::NumericTablePtr convert_to_daal_table(const table& table) {
     if (table.get_kind() == homogen_table::kind()) {
         const auto& homogen = static_cast<const homogen_table&>(table);
-        return convert_to_daal_table<Float>(homogen);
+        return convert_to_daal_table<Data>(homogen);
     }
     else {
-        return copy_to_daal_homogen_table<Float>(table);
+        return copy_to_daal_homogen_table<Data>(table);
     }
 }
 
 #ifdef ONEDAL_DATA_PARALLEL
-template <typename T>
-inline auto convert_to_daal_sycl_homogen_table(sycl::queue& queue,
-                                               array<T>& data,
-                                               std::int64_t row_count,
-                                               std::int64_t column_count) {
-    data.need_mutable_data(queue);
-    const auto daal_data =
-        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_object_owner{ data });
+inline daal::data_management::NumericTablePtr convert_to_daal_table(const sycl::queue& queue,
+                                                                    const table& table) {
+    return interop::sycl_table_adapter::create(queue, table);
```

Comment:
Why the logic differs from host version of that function?

Reply:
Because `sycl_table_adapter` is not DPC++ equivalent of  `host_homogen_table_adapter`. 

`sycl_table_adapter` can wrap arbitrary table and extract data using `row_accessor` or `column_accessor`, but `host_homogen_table_adapter` wraps only `homogen_table`. I prefer to use old SYCL entities as little as possible to fix possible functional or perf. problems on side of new interfaces.